worlds/saving_princess/Rules.py

```diff
@@ -0,0 +1,110 @@
+from BaseClasses import CollectionState, Location
+from worlds.generic.Rules import set_rule
+
+from .Constants import *
+
+
+def set_rules(world: "SavingPrincessWorld"):
+    def get_location(name: str) -> Location:
+        return world.multiworld.get_location(name, world.player)
+
+    def get_region_entrance(name: str) -> Location:
+        return world.multiworld.get_entrance(f"{name} entrance", world.player)
+
+    def can_hover(state: CollectionState) -> bool:
+        # portia can hover if she has a weapon other than the powered blaster and 4 reload speed upgrades
+        return (
+                state.has(ITEM_RELOAD_SPEED, world.player, 4)
+                and (
+                        state.has(ITEM_WEAPON_FIRE, world.player)
+                        or state.has(ITEM_WEAPON_ICE, world.player)
+                        or state.has(ITEM_WEAPON_VOLT, world.player)
```

Comment:
```suggestion
                        state.has_any({ITEM_WEAPON_FIRE, ITEM_WEAPON_ICE, ITEM_WEAPON_VOLT}, world.player)
```
You can use state.has_any to do basically what you're doing here